src/codegen/encoding/mod.rs

```diff
@@ -42,17 +535,30 @@ pub(super) trait AbiEncoding {
         buffer_size: Option<Expression>,
     ) -> Vec<Expression>;
 
-    /// Cache items loaded from storage to reuse them later, so we avoid the expensive operation
-    /// of loading from storage twice. We need the data in two different passes: first to
-    /// calculate its size and then to copy it to the buffer.
+    /// Encoding happens in two steps. First, we look at each argument to calculate its size. If an
+    /// argument is a storage variable, we load it and save it to a local variable.
+    ///
+    /// During a second pass, we copy each argument to a buffer. To copy storage variables properly into
+    /// the buffer, we must load them from storage and save them in a local variable. As we have
+    /// already done this before, we can cache the Expression::Variable, containing the items we loaded before.
```

Comment:
```suggestion
    /// already done this, we can cache the Expression::Variable, containing the items we loaded before.
```
`before` is repeated.